operators/src/adapters/raster_time.rs

```diff
@@ -16,45 +16,59 @@ use std::task::{Context, Poll};
 /// Assumes that raster tiles are contiguous temporally, with no-data-tiles filling gaps.
 /// Potentially queries the same tiles multiple times from its sources.
 #[pin_project(project = RasterTimeAdapterProjection)]
-pub struct RasterTimeAdapter<T1, T2, St1, St2, F1, F2>
+pub struct RasterTimeAdapter<T1, T2, F1, F2>
 where
     T1: Pixel,
     T2: Pixel,
-    St1: Stream<Item = Result<RasterTile2D<T1>>>,
-    St2: Stream<Item = Result<RasterTile2D<T2>>>,
-    F1: Fn(RasterQueryRectangle) -> St1,
-    F2: Fn(RasterQueryRectangle) -> St2,
+    F1: Queryable<T1>,
+    F2: Queryable<T2>,
+    F1::Stream: Stream<Item = Result<RasterTile2D<T1>>>,
+    F2::Stream: Stream<Item = Result<RasterTile2D<T2>>>,
+    F1::Output: Future<Output = Result<F1::Stream>>,
+    F2::Output: Future<Output = Result<F2::Stream>>,
 {
+    // the first source (wrapped QueryProcessor)
     source_a: F1,
+    // the second source (wrapped QueryProcessor)
     source_b: F2,
+    #[pin]
+    // place for storing the output of the `query` method of the two sources
+    // and "awaiting" it
+    query_a_b_fut: Option<Join<F1::Output, F2::Output>>,
```

Comment:
I think that is good and viable to extend to use a tuple of sources and a (already) tuple out joined/zipped tiles.

Reply:
was that a compliment or a suggestion? Please elaborate :D

Reply:
Yeah, well, this is a good first step. Then, we should use tuples for the sources and implement it for 2 to 12 sources or something like this.